src/DotNetWorker.ApplicationInsights/FunctionsTelemetryModule.cs

```diff
@@ -23,15 +23,15 @@ public void Initialize(TelemetryConfiguration configuration)
                 ShouldListenTo = source => source.Name.StartsWith("Microsoft.Azure.Functions.Worker"),
                 ActivityStarted = activity =>
                 {
-                    var dependency = new DependencyTelemetry("Azure.Functions", activity.OperationName, activity.OperationName, null);
+                    var dependency = _telemetryClient.StartOperation<DependencyTelemetry>(activity);
+                    dependency.Telemetry.Type = "Azure.Functions";
                     activity.SetCustomProperty("_depTel", dependency);
-                    dependency.Start();
                 },
                 ActivityStopped = activity =>
                 {
-                    var dependency = activity.GetCustomProperty("_depTel") as DependencyTelemetry;
-                    dependency.Stop();
-                    _telemetryClient.TrackDependency(dependency);
+                    var dependency = activity.GetCustomProperty("_depTel") as IOperationHolder<DependencyTelemetry>;
```

Comment:
can we use constants for all the strings here?

Reply:
Yes, I'll catch that in the next rev. I'm still not completely convinced this is the write way to store the telemetry throughout the Activity. I'll need to follow up with experts.